src/sorting/counting_sort.rs

```diff
@@ -0,0 +1,44 @@
+/// In place counting sort for collections of u32
+/// O(n + maxval) in time, where maxval is the biggest value an input can possibly take
+/// O(maxval) in memory
+/// u32 is chosen arbitrarly, a counting sort probably should'nt be used on data that requires bigger types.
+pub fn counting_sort(arr: &mut [u32], maxval: usize) {
+    let mut occurences: Vec<usize> = vec![0; maxval + 1];
+
+    for &data in arr.iter() {
+        occurences[data as usize] += 1;
```

Comment:
I was just thinking that we can we use counting sort, if we have negative numbers in the array?